internal/data/sql/queries/metrics.sql

```diff
@@ -62,7 +68,35 @@ INSERT into block(
 	block_hash,
 	block_number,
 	block_timestamp,
-	tx_hash
+	tx_hash,
+	slot
+) 
+VALUES ($1, $2, $3, $4, $5) 
+ON CONFLICT DO NOTHING;
+
+-- name: QueryBlockFromSlot :one
+SELECT * FROM block
+WHERE slot = $1 FOR UPDATE;
```

Comment:
Looks like this query is unused. 

But if it's not I'd suggest to include the fact that it's locking for update in the name to prevent accidental use in a context where no lock is required.